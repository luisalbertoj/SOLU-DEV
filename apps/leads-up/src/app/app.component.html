<input type="text" [(ngModel)]="message.text" name="message.text" />
<button (click)="sendMessage()">send</button>
<a routerLink="/">Home</a>
<a routerLink="/auth/login">Login</a>
<a routerLink="/auth/register">Register</a>

<router-outlet></router-outlet>
