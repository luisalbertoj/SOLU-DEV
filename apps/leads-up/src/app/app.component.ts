import { Component } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Message } from '@solu-dev/api-interfaces';

@Component({
  selector: 'solu-dev-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css'],
})
export class AppComponent {
  hello$ = this.http.get<Message>('/api/hello');
  constructor(private http: HttpClient) {}
}
