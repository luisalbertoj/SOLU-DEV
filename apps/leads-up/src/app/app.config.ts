import { Injectable } from '@angular/core';
import { JwtHelperService } from '@auth0/angular-jwt';

const helper = new JwtHelperService();

@Injectable()
export class AppConfig {
  load() {
    return new Promise((resolve) => {
      const token =
        localStorage.getItem('auth_token') === null
          ? false
          : localStorage.getItem('auth_token');
      if (token) {
        try {
          if (helper.isTokenExpired(token) === true) {
            localStorage.removeItem('auth_token');
          }
        } catch (exception) {
          localStorage.removeItem('auth_token');
        }
      }
      resolve(true);
    });
  }
}
