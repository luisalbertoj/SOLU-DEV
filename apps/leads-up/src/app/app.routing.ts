import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

const routes: Routes = [
  {
    path: 'auth',
    title: 'auth',
    loadChildren: () => import('./auth/auth.module').then((m) => m.AuthModule),
  },
  { path: '**', redirectTo: '/auth', pathMatch: 'full' },
];

@NgModule({
  imports: [
    RouterModule.forRoot(routes, { scrollPositionRestoration: 'enabled' }),
  ],
  exports: [RouterModule],
})
export class AppRoutingModule {}
