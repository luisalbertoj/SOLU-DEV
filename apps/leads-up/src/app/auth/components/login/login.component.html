<div class="login-body">
  <div class="login-image">
    <!-- <img
      [src]="
        'assets/layout/images/pages/login-' +
        (app.colorScheme === 'dark' ? 'ondark' : 'onlight') +
        '.png'
      "
      alt="atlantis"
    /> -->
  </div>
  <div class="login-panel p-fluid">
    <div class="flex flex-column">
      <div class="flex align-items-center mb-6 logo-container">
        <img
          [src]="
            'assets/layout/images/logo-' +
            (app.colorScheme === 'light' ? 'dark' : 'light') +
            '.png'
          "
          class="login-logo"
        />
        <img
          [src]="
            'assets/layout/images/appname-' +
            (app.colorScheme === 'light' ? 'dark' : 'light') +
            '.png'
          "
          class="login-appname"
        />
      </div>
      <div class="form-container">
        <span class="p-input-icon-left">
          <i class="pi pi-user"></i>
          <solu-dev-input-text-one
            type="text"
            autocomplete="off"
            placeholder="Username"
            [(model)]="loginData.username"
            class="p-input-icon-left"
          ></solu-dev-input-text-one>
        </span>
        <span class="p-input-icon-left">
          <i class="pi pi-key"></i>
          <solu-dev-input-text-one
            type="password"
            autocomplete="off"
            placeholder="Password"
            [(model)]="loginData.password"
            class="p-input-icon-left"
          ></solu-dev-input-text-one>
        </span>
        <a class="flex">Forgot your password?</a>
      </div>
      <div class="button-container">
        <solu-dev-button-one
          type="button"
          textContent="Login"
          (clickButton)="handleLogin()"
        ></solu-dev-button-one>
      </div>
    </div>

    <div class="login-footer flex align-items-center">
      <div class="flex align-items-center login-footer-logo-container">
        <img
          src="assets/layout/images/logo-gray.png"
          class="login-footer-logo"
        />
        <img
          src="assets/layout/images/appname-gray.png"
          class="login-footer-appname"
        />
      </div>
      <span>Copyright 2022</span>
    </div>
  </div>
</div>
