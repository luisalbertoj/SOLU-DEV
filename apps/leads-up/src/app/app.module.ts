import { HashLocationStrategy, LocationStrategy } from '@angular/common';
import { HttpClientModule, HTTP_INTERCEPTORS } from '@angular/common/http';
import { APP_INITIALIZER, NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { TitleStrategy } from '@angular/router';
import { NgxsReduxDevtoolsPluginModule } from '@ngxs/devtools-plugin';
import { NgxsLoggerPluginModule } from '@ngxs/logger-plugin';
import { NgxsModule } from '@ngxs/store';
import { MessageService } from 'primeng/api';
import { environment } from '../environments/environment';
import { AppComponent } from './app.component';
import { AppConfig } from './app.config';
import { AppRoutingModule } from './app.routing';
import { AccessInterceptor } from './core/interceptors/access.interceptor.service';
import { AuthInterceptor } from './core/interceptors/auth.interceptor.service';
import { PageTitleStrategy } from './core/services/page-title.service';
import { TokenProvider } from './core/services/token.service';
import { ToastService } from './shared/toast/services/toast.service';
import { ToastComponent } from './shared/toast/toast.component';
import { MessagesState } from './store/chat/chat.state';

export function servicesOnRun(config: AppConfig, token: TokenProvider) {
  return () => config.load().then(() => token.load());
}
@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    AppRoutingModule,
    HttpClientModule,
    NgxsModule.forRoot([], {
      developmentMode: !environment.production,
    }),
    NgxsReduxDevtoolsPluginModule.forRoot({
      disabled: environment.production,
    }),
    NgxsLoggerPluginModule.forRoot({ disabled: environment.production }),
    NgxsModule.forRoot([MessagesState]),
    ToastComponent,
  ],
  providers: [
    AppConfig,
    TokenProvider,
    {
      provide: APP_INITIALIZER,
      useFactory: servicesOnRun,
      multi: true,
      deps: [AppConfig, TokenProvider],
    },
    { provide: HTTP_INTERCEPTORS, useClass: AuthInterceptor, multi: true },
    { provide: HTTP_INTERCEPTORS, useClass: AccessInterceptor, multi: true },
    {
      provide: TitleStrategy,
      useClass: PageTitleStrategy,
    },
    {provide: LocationStrategy, useClass: HashLocationStrategy},
    ToastService,
    MessageService,
  ],
  bootstrap: [AppComponent],
})
export class AppModule {}
