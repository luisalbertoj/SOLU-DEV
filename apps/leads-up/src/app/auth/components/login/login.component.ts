import { Component } from '@angular/core';
import { LoginModel, LoginResModel } from '@app/auth/models/login.model';
import { LoginService } from '@app/auth/services/login.service';
import { Store } from '@ngxs/store';
import { Observable } from 'rxjs';

@Component({
  selector: 'solu-dev-login',
  templateUrl: './login.component.html',
  styleUrls: ['./login.component.css'],
})
export class LoginComponent {
  public loginData: LoginModel;
  messages: Observable<any>;
  constructor(private _loginService: LoginService, private _store: Store) {
    this.loginData = new LoginModel();
    this.messages = this._store.select((state) => state.messages.messages);
  }

  find(): void {
    this._loginService.find(this.loginData).subscribe({
      next: (response: LoginResModel) => {
        return console.log({ response });
      },
      error: (error) => console.log({ error }),
    });
  }
}
