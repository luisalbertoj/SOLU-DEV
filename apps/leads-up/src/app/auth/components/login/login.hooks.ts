import { LoginModel } from '../../models/login.model';

export class LoginHooks {
  static validateData({ username, password }: LoginModel): boolean {
    if (!password || !username) {
      return false;
    }
    return true;
  }
}
