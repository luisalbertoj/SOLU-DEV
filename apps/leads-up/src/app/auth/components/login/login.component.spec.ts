import { ComponentFixture, TestBed } from '@angular/core/testing';

import {
  HttpClientTestingModule,
  HttpTestingController,
} from '@angular/common/http/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { MessageService } from 'primeng/api';
import { take } from 'rxjs';
import { testData } from '../../../../assets/test-data';
import { environment } from '../../../../environments/environment';
import { AppComponent } from '../../../app.component';
import { ToastService } from '../../../shared/toast/services/toast.service';
import { LoginService } from '../../services/login.service';
import { LoginComponent } from './login.component';
import { LoginHooks } from './login.hooks';

describe('LoginComponent', () => {
  let component: LoginComponent;
  let fixture: ComponentFixture<LoginComponent>;
  let compiled: HTMLElement;
  let http: HttpTestingController;
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [],
      imports: [HttpClientTestingModule, LoginComponent, RouterTestingModule],
      providers: [LoginService, AppComponent, ToastService, MessageService],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(LoginComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
    http = TestBed.inject(HttpTestingController);
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('this should be equal to the snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  });
  it('should be validate if data is invalid', () => {
    expect(LoginHooks.validateData(testData.loginCase.case2)).toBe(false);
  });
  it('should be validate if data is valid', () => {
    expect(LoginHooks.validateData(testData.loginCase.case1)).toBe(true);
  });
  it('should be validate if data is invalid called login', () => {
    component.loginData = testData.loginCase.case2;
    expect(component.handleLogin(new Event('', undefined))).toBe(false);
  });
  it('this should validate the login method', (done) => {
    component.loginData = testData.loginCase.case1;
    component.isLoading$
      .pipe(take(1))
      .subscribe((loading) => expect(loading).toBe(false));
    expect(component.handleLogin(new Event('', undefined))).toBe(true);

    const req = http.expectOne(environment.apiRoute + 'auth/login');
    expect(req.request.method).toBe('POST');
    expect(JSON.stringify(req.request.body)).toBe(
      JSON.stringify(component.loginData)
    );
    done();
  });
});
