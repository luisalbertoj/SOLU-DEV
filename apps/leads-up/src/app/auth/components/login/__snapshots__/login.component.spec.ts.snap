// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent this should be equal to the snapshot 1`] = `"<div class=\\"login-body\\"><div class=\\"login-image\\"></div><div class=\\"login-panel p-fluid\\"><div class=\\"flex flex-column\\"><div class=\\"flex align-items-center mb-6 logo-container\\"><img class=\\"login-logo\\" src=\\"assets/layout/images/logo-light.png\\"><img class=\\"login-appname\\" src=\\"assets/layout/images/appname-light.png\\"></div><form><div class=\\"form-container\\"><span class=\\"p-input-icon-left\\"><i class=\\"pi pi-user\\"></i><solu-dev-input-text-one type=\\"text\\" autocomplete=\\"off\\" placeholder=\\"Username\\" name=\\"username\\" class=\\"p-input-icon-left\\" ng-reflect-type=\\"text\\" ng-reflect-autocomplete=\\"off\\" ng-reflect-placeholder=\\"Username\\" ng-reflect-name=\\"username\\" ng-reflect-required=\\"true\\" ng-reflect-model=\\"\\" ng-reflect-autofocus=\\"true\\"><input pinputtext=\\"\\" autofocus=\\"\\" class=\\"p-inputtext p-component p-element ng-invalid ng-untouched ng-pristine\\" type=\\"text\\" autocomplete=\\"off\\" placeholder=\\"Username\\" ng-reflect-model=\\"\\" ng-reflect-ng-class=\\"ng-dirty ng-invalid\\" ng-reflect-required=\\"true\\" ng-reflect-name=\\"username\\" required=\\"\\"></solu-dev-input-text-one></span><span class=\\"p-input-icon-left\\"><i class=\\"pi pi-key\\"></i><solu-dev-input-text-one type=\\"password\\" autocomplete=\\"off\\" name=\\"password\\" placeholder=\\"Password\\" class=\\"p-input-icon-left\\" ng-reflect-type=\\"password\\" ng-reflect-autocomplete=\\"off\\" ng-reflect-name=\\"password\\" ng-reflect-placeholder=\\"Password\\" ng-reflect-required=\\"true\\" ng-reflect-model=\\"\\"><input pinputtext=\\"\\" class=\\"p-inputtext p-component p-element ng-invalid ng-untouched ng-pristine\\" type=\\"password\\" autocomplete=\\"off\\" placeholder=\\"Password\\" ng-reflect-model=\\"\\" ng-reflect-ng-class=\\"ng-dirty ng-invalid\\" ng-reflect-required=\\"true\\" ng-reflect-name=\\"password\\" required=\\"\\"></solu-dev-input-text-one></span><a class=\\"flex\\">Forgot your password?</a></div><div class=\\"button-container\\"><solu-dev-button-one type=\\"submit\\" textcontent=\\"Login\\" ng-reflect-type=\\"submit\\" ng-reflect-text-content=\\"Login\\"><button pbutton=\\"\\" pripple=\\"\\" class=\\"p-element p-ripple p-button p-component\\" type=\\"submit\\" ng-reflect-label=\\"Login\\"><span class=\\"p-button-label\\">Login</span><span class=\\"p-ink\\"></span></button></solu-dev-button-one></div></form></div><div class=\\"login-footer flex align-items-center\\"><div class=\\"flex align-items-center login-footer-logo-container\\"><img src=\\"assets/layout/images/logo-gray.png\\" class=\\"login-footer-logo\\"><img src=\\"assets/layout/images/appname-gray.png\\" class=\\"login-footer-appname\\"></div><span>Copyright 2022</span></div></div></div>"`;
